src/lib/containers/table/datasets/DatasetItemsEditor.tsx

```diff
@@ -160,60 +180,91 @@ export function DatasetItemsEditor(props: DatasetItemsEditorProps) {
     }
   })
 
+  function getDataSetDifference(
+    oldDataSet: EntityRef[],
+    newDataSet: EntityRef[],
+  ) {
+    let unchangedItems = oldDataSet.filter(
+      oldItem =>
+        !newDataSet.find(newItem => newItem.entityId === oldItem.entityId),
+    )
+    const deletedItems = [...unchangedItems]
+    const { updatedItems, newItems } = newDataSet.reduce(
+      (results, result) => {
+        if (
+          oldDataSet.find(
+            old =>
+              old.entityId === result.entityId &&
+              old.versionNumber !== result.versionNumber,
+          )
+        ) {
+          results['updatedItems'].push(result)
+        }
+        if (!oldDataSet.find(old => old.entityId === result.entityId)) {
+          results['newItems'].push(result)
+        }
+        if (
+          oldDataSet.find(
+            old =>
+              old.entityId === result.entityId &&
+              old.versionNumber === result.versionNumber,
+          )
+        ) {
+          unchangedItems.push(result)
+        }
```

Comment:
These are all mutually exclusive cases so instead of going through oldDataSet 3 times, can you just do this?

```suggestion
        const oldItem = oldDataSet.find(old => old.entityId === result.entityId)
        if (oldItem) {
          if (result.versionNumber !== old.versionNumber) {
            results['updatedItems'].push(result)
          } else {
            unchangedItems.push(result)
          }
        } else {
          results['newItems'].push(result)
	}
```